src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```diff
@@ -0,0 +1,248 @@
+     * > or before the `init` call, using the `options` top-level function (see
+     * > [Components - Passing Component Options Before the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsBeforetheinitCall)).
+     *
+     * **Example:**
+     *
+     * ```javascript
+     *
+     * var suggestionTemplate = function(row, omnibox) {
+     *   return "Searching for " + row.keyword + " in category " + row.value;
+     * };
+     *
+     * // You can set the option in the 'init' call:
+     * Coveo.init(document.querySelector("#search"), {
+     *    FacetValueSuggestions : {
+     *      templateHelper : suggestionTemplate
+     *    }
+     * });
+     *
+     * // Or before the 'init' call, using the 'options' top-level function:
+     * // Coveo.options(document.querySelector("#search"), {
+     * //   FacetValueSuggestions : {
+     * //     templateHelper : suggestionTemplate
+     * //   }
+     * // });
+     * ```
+     */
+    templateHelper: ComponentOptions.buildCustomOption<(row: IFacetValueSuggestionRow, omnibox: Omnibox) => string>(() => {
+      return null;
+    })
+  };
+
+  public fieldValueCache: SuggestionsCache<IFacetValueSuggestionRow[]> = new SuggestionsCache();
+
+  public facetValueSuggestionsProvider: IFacetValueSuggestionsProvider;
+
+  static defaultTemplate(this: FacetValueSuggestions, row: IFacetValueSuggestionRow, omnibox: Omnibox): string {
+    const keyword = DomUtils.highlightElement(row.keyword, omnibox.getText(), 'coveo-omnibox-hightlight2');
+    const facetValue = DomUtils.highlightElement(row.value, row.value, 'coveo-omnibox-hightlight');
+    const details = this.options.displayEstimateNumberOfResults ? ` (${row.numberOfResults} results)` : '';
+    return `${keyword} in ${facetValue}${details}`;
+  }
+
+  /**
+   * Creates a new `FieldSuggestions` component.
+   * @param element The HTMLElement on which to instantiate the component.
+   * @param options The options for the `FieldSuggestions` component.
+   * @param bindings The bindings that the component requires to function normally. If not set, these will be
+   * automatically resolved (with a slower execution time).
+   */
+  constructor(element: HTMLElement, public options: IFacetValueSuggestionsOptions, bindings?: IComponentBindings) {
+    super(element, FacetValueSuggestions.ID, bindings);
+
+    this.facetValueSuggestionsProvider = new FacetValueSuggestionsProvider(this.queryController, this.queryStateModel, {
+      field: <string>this.options.field
+    });
+
+    this.options = ComponentOptions.initComponentOptions(element, FacetValueSuggestions, options);
+    if (!this.options.templateHelper) {
+      this.options.templateHelper = FacetValueSuggestions.defaultTemplate;
+    }
+
+    Assert.check(Utils.isCoveoField(<string>this.options.field), `${this.options.field} is not a valid field`);
```

Comment:
This check is already made when loading a field option. I do not believe this is necessary.